mlflow/pyfunc/model.py

```diff
@@ -73,16 +74,57 @@ def _log_warning_if_params_not_in_predict_signature(logger, params):
         )
 
 
-class PythonModel:
+class _PythonModelMetaclass(ABCMeta):
+    def __call__(cls, *args, **kwargs):
+        cls._warn_on_setting_model_in_init()
+        return super().__call__(*args, **kwargs)
+
+    def _warn_on_setting_model_in_init(cls):
+        if not cls.__bases__ == (PythonModel,):
+            return
+
+        import click
+
+        message = (
+            click.style(
+                "It looks like you're trying to save a model as an instance attribute. ",
```

Comment:
Can we include example code in the message?

Reply:
yeah that sounds good!